hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+			}
+			else
+			{
+			ROS_ERROR_STREAM("Could not solve for a valid path using the Descartes dense planner");
+			exit(-1);
+			}
+
+			succeeded = dense_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+
+		if(config_.descartes_planner_type == "sparse")
+		{
+			bool succeeded = sparse_planner_.planPath(descartes_traj_input_);
+			if (succeeded)
+			{
+			ROS_INFO_STREAM("Valid path was found using the Descartes sparse planner");
+			}
+			else
+			{
+			ROS_ERROR_STREAM("Could not solve for a valid path using the Descartes sparse planner");
+			exit(-1);
+			}
+
+			succeeded = sparse_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+		
+		ROS_INFO_STREAM("Task '"<<__FUNCTION__<<"' completed");
+	}
+
+	void HybridPlanner::runPath()
+	{
+		moveit_group_ptr_->setPlannerId(PLANNER_ID);//TOCHECK should be in init(). Also the config_ param or not doubt. Check header file. 
+		overall_robot_traj_.getRobotTrajectoryMsg(moveit_robot_traj_);
+
+		// creating goal joint pose to start of the path
+		// std::vector<double> seed_pose(robot_model_ptr_->getDOF());
+  		std::vector<double> start_pose;
+  		descartes_core::TrajectoryPtPtr first_point_ptr = path[0];
+		first_point_ptr->getNominalJointPose(config_.seed_pose,*descartes_robot_model_ptr_,start_pose);
+
+		// moving arm to joint goal
+		moveit_group_ptr_->setJointValueTarget(start_pose);
+		moveit_group_ptr_->setPlanningTime(config_.plan_time);
+		moveit_msgs::MoveItErrorCodes result = moveit_group_ptr_->move();
+		if(result.val != result.SUCCESS)
+		{
+		ROS_ERROR_STREAM("Move to start joint pose failed");
+		exit(-1);
+		}
+
+		// creating Moveit! joint trajectory from Descartes Trajectory
+		fromDescartesToMoveitTrajectory();
+
+		//all free space segments now in moveit_robot_traj_ and 
+		//all process segments in trajectory_msgs::JointTrajectory& joint_traj_descartes_
+		//TOCHECK TODO to combine them, again use robot_trajectory?
+	}
+
+	void HybridPlanner::fromDescartesToMoveitTrajectory()
+	{
+	  // Fill out information about our trajectory
+	  joint_traj_descartes_.header.stamp = ros::Time::now();
+	  joint_traj_descartes_.header.frame_id = config_.world_frame;
+	  joint_traj_descartes_.joint_names = config_.joint_names;
+
+	  // For keeping track of time-so-far in the trajectory
+	  double time_offset = 0.0;
+
+	  // Loop through the trajectory
+	  for (unsigned int i = 0; i < descartes_traj_input_.size(); i++)
+	  {
+	    // Find nominal joint solution at this point
+	    std::vector<double> joints;
+
+	    // getting joint position at current point
+	    const descartes_core::TrajectoryPtPtr& joint_point = descartes_traj_input_[i];
+	    joint_point->getNominalJointPose(std::vector<double>(), *descartes_robot_model_ptr_, joints);
```

Comment:
missing param? seeds? what does nominal mean?
`00106 bool JointTrajectoryPt::getNominalJointPose(const std::vector<double> &seed_state,`
`00107                                             const RobotModel &model,`
`00108                                             std::vector<double> &joint_pose) const`
`00109 {`
`00110   joint_pose.resize(joint_position_.size());`
`00111   for (size_t ii=0; ii<joint_position_.size(); ++ii)`
`00112   {`
`00113     joint_pose[ii] = joint_position_[ii].nominal;`
`00114   }`
`00115   return true;`
`00116 }`